### ISSUE TITLE
[Bug] windows run build thread 'main' has overflowed its stack

### ISSUE THREAD

**Issue by daodao2007** [move-language/move#372]
# 🐛 Bug

**Stack trace/error message**
```
target\debug\move -d -v build
INCLUDING DEPENDENCY MoveStdlib
BUILDING hello001

thread 'main' has overflowed its stack
```

## System information

**Please complete the following information:**
-  Move Version 
```
..\target\debug\move -V
move-cli 0.1.0
```
-  Rust Version 
``` 
rustc -V
rustc 1.62.0 (a8314ef7d 2022-06-27)
``` 
-  Computer OS
windows 11

**Comment by vgao1996** [comment]
Interesting... I know that the default stack size on windows is smaller than that on linux, but still curious what caused this to happen.

**Comment by vgao1996** [comment]
It seems like the stack size of the main thread is set by the OS so it might be hard to change that. However, we may be able to spawn a child thread and use that as the main thread instead:
https://stackoverflow.com/questions/44003589/how-to-increase-the-stack-size-available-to-a-rust-library

Still, we need to investigate what caused the stack overflow. I'd be a bit surprised we're actually using up 1 MB of stack space.

**Comment by awelc** [comment]
> However, we may be able to spawn a child thread and use that as the main thread instead: https://stackoverflow.com/questions/44003589/how-to-increase-the-stack-size-available-to-a-rust-library

I had to do it already for the Move analyzer: https://github.com/move-language/move/blob/f6490c1d13b5ba36c4d0c7b9724ac929b46cfac2/language/move-analyzer/src/symbols.rs#L365

**Comment by awelc** [comment]
More generally, if we want to test Windows build in CI we also need to adjust the tests (at least) - comparison of expected output with generated output fails on Windows as well.

**Comment by daodao2007** [comment]
You're right. I won't have this problem if I change it this way.
```
//language\tools\move-cli\src\main.rs
fn main() -> Result<()> {
    let builder = thread::Builder::new()
        .name("move_cli".into())
        .stack_size(32 * 1024 * 1024); // 32MB of stack space

    let handler = builder.spawn(|| {
        let error_descriptions: ErrorMapping = bcs::from_bytes(move_stdlib::error_descriptions())?;
        let cost_table = &move_vm_test_utils::gas_schedule::INITIAL_COST_SCHEDULE;
        let addr = AccountAddress::from_hex_literal("0x1").unwrap();
        let natives = all_natives(addr, GasParameters::zeros())
            .into_iter()
            .chain(nursery_natives(addr, NurseryGasParameters::zeros()))
            .collect();

        move_cli::move_cli(natives, cost_table, &error_descriptions)
    }).unwrap();

    handler.join().unwrap();
    Ok(())
}
```

**Comment by LemonHX** [comment]
In starcoin we have fixed by ajusting the stack size in compile time

**Comment by LemonHX** [comment]
> It seems like the stack size of the main thread is set by the OS so it might be hard to change that. However, we may be able to spawn a child thread and use that as the main thread instead: https://stackoverflow.com/questions/44003589/how-to-increase-the-stack-size-available-to-a-rust-library
> 
> Still, we need to investigate what caused the stack overflow. I'd be a bit surprised we're actually using up 1 MB of stack space.

and no we don't need to, we just modify the compile settings

**Comment by LemonHX** [comment]
![图片](https://user-images.githubusercontent.com/35558393/184327023-b9dbf7ed-fbea-4f67-9aba-d9c2fdb275fa.png)
fixed in pr #375

**Comment by awelc** [comment]
> and no we don't need to, we just modify the compile settings

I haven't checked it myself, but one of our engineers reported that this does not work (for our chain) when the user installs the binary via `cargo install --locked --git https://...`
